Core/Core/Store/UseCase.swift

```diff
@@ -63,6 +63,9 @@ extension UseCase {
     }
 
     public func fetch(environment: AppEnvironment = .shared, force: Bool = false, _ callback: @escaping RequestCallback) {
+        // Make sure we write to the database that initiated this request
+        let database = environment.database
```

Comment:
i'm lost here, was another environment being used than the one passed in?

Reply:
The `environment` doesn't change but `environment.database` does change between the time the `fetch` starts and completes.